pkg/storage/utils/decomposedfs/node/node.go

```diff
@@ -888,7 +887,7 @@ func (n *Node) GetTMTime(ctx context.Context) (time.Time, error) {
 
 // GetMTime reads the mtime from disk
 func (n *Node) GetMTime() (time.Time, error) {
-	fi, err := os.Lstat(n.InternalPath())
+	fi, err := os.Stat(n.lu.MetadataBackend().MetadataPath(n.InternalPath()))
```

Comment:
could we store the mtime in the metadata itself? we propably got the metadata for the node already anyway so sparing the roundtrip to the storage would be desirable. Would also simplify propagating the tree size and tree modification time into one operation.

Reply:
we certainly can and should. I just did not go in that direction first, because I stumbled over the complexity of moving all files and folders belonging to a node into trash: the dir (or unneded file) with children, the metadata file and maybe lock files?